**# 1. Mendelian randomization analysis (IVW and other complementary studies, e.g. MRPRESSO and MRlap) and Multivariable Mendelian randomization**



library(TwoSampleMR)

library(MendelianRandomization)

library(data.table)



**# 1.1 Performing Mendelian Randomization study (¡°IOP to IS¡± as an example, similar process # can be applied to other traits of interests)**

exposure\_iop\_r<-extract\_instruments('ukb-b-14146')

exposure\_dat4 <- format\_data(

&nbsp; exposure\_iop\_r,

&nbsp; type = "exposure",

&nbsp; snps = NULL,

&nbsp; header = TRUE,

&nbsp; snp\_col = "SNP",

&nbsp; beta\_col = "beta.exposure",

&nbsp; se\_col = "se.exposure",

&nbsp; eaf\_col = "eaf.exposure",

&nbsp; effect\_allele\_col = "effect\_allele.exposure",

&nbsp; other\_allele\_col = "other\_allele.exposure",

&nbsp; pval\_col = "pval.exposure",

&nbsp; samplesize\_col = "samplesize.exposure",

&nbsp; gene\_col = "genes",

&nbsp; id\_col = "id.exposure",

&nbsp; min\_pval = 1e-200,

&nbsp; chr\_col = "chr.exposure",

&nbsp; pos\_col = "pos.exposure",

&nbsp; log\_pval = FALSE,phenotype\_col = 'id.exposure'

)

colnames(exposure\_dat4)\[7]<-'rsid'

colnames(exposure\_dat4)\[5]<-'pval'

exposure\_dat5<-ld\_clump(

&nbsp;         dat = exposure\_dat4,

&nbsp;         clump\_kb = 10000,

&nbsp;         clump\_r2 = 0.001,

&nbsp;         pop = "EUR"

&nbsp;     )

&nbsp;colnames(exposure\_dat5)\[7]<-'SNP'

&nbsp;colnames(exposure\_dat5)\[5]<-'pval.exposure'

&nbsp;exposure\_dat6<-exposure\_dat5

&nbsp;chd\_out\_dat <- extract\_outcome\_data(

&nbsp;    snps = exposure\_dat6$SNP,

&nbsp;    outcomes = 'ebi-a-GCST006908',proxies = 1, rsq = 0.8, align\_alleles = 1, palindromes = 1, maf\_threshold = 0.3

&nbsp;)

dat <- harmonise\_data(

&nbsp;   exposure\_dat = exposure\_dat6, 

&nbsp;   outcome\_dat = chd\_out\_dat

)

res <- mr(dat)

iop\_is\_heterogeneity<-mr\_heterogeneity(dat)

iop\_is\_pleiotrophy<-mr\_pleiotropy\_test(dat)

p1 <-mr\_scatter\_plot(res, dat)

iop\_is\_leaveoneout<-mr\_leaveoneout(dat)

p2<-mr\_leaveoneout\_plot(iop\_is\_leaveoneout)



**# 1.2 Performing MRPRESSO analysis**

library(MRPRESSO)

mr\_presso<-mr\_presso(BetaOutcome ="beta.outcome", BetaExposure = "beta.exposure", SdOutcome ="se.outcome", SdExposure = "se.exposure", OUTLIERtest = TRUE,DISTORTIONtest = TRUE, data = dat, NbDistribution = 1000,  SignifThreshold = 0.05)

capture.output(mr\_presso,file="iop\_is\_mr\_presso.txt")



**# 1.3 Performing MRlap analysis** 

**# download finn-b-H7\_GLAUCPRIMOPEN.vcf.gz, ukb-b-14146.vcf.gz and ebi-a-GCST006908.vcf.gz files from the IEU GWAS platform.** 

**# download eur\_w\_ld\_chr and w\_hm3.noMHC.snplist files according to the MRlap GitHub repository.**



library(MRlap)

library(data.table)

library(stringr)

library(tidyr)



**# import summary statistics data**

poag<-fread('./finn-b-H7\_GLAUCPRIMOPEN.vcf.gz', fill=TRUE)

iop<-fread('./ukb-b-14146.vcf.gz', fill=TRUE)

is<-fread('./ebi-a-GCST006908.vcf.gz', fill=TRUE)



**# preparation of POAG data** 

poag<-poag\[109:16278346 ,]

colnames(poag)="NAME"

&nbsp;poag\_split<-str\_split\_fixed(poag$NAME, '\\t', 10)

poag\_split<-poag\_split\[,c(1:5,10)]

poag\_split<-data.frame(poag\_split)

poag\_split1<-poag\_split\[,6]

poag\_split1<-str\_split\_fixed(poag\_split$X6, ':', 5)

poag\_split1<-data.frame(poag\_split1)

&nbsp;poag\_combine<-cbind(poag\_split\[,1:5],poag\_split1)

poag\_combine<-poag\_combine\[,1:9]

colnames(poag\_combine)<-c('chr','pos','rsid','ref','alt',"beta","se","pval","maf")

poag\_combine<-poag\_combine\[-1,]

&nbsp;poag\_combine\[,1]<-as.numeric(poag\_combine\[,1])

poag\_combine\[,2]<-as.numeric(poag\_combine\[,2])

poag\_combine\[,6]<-as.numeric(poag\_combine\[,6])

poag\_combine\[,7]<-as.numeric(poag\_combine\[,7])

&nbsp;poag\_combine\[,8]<-as.numeric(poag\_combine\[,8])

&nbsp;poag\_combine\[,9]<-as.numeric(poag\_combine\[,9])

poag\_combine$pval<-10^(-poag\_combine$pval)

poag\_combine$maf<-ifelse(poag\_combine$maf>0.5,poag\_combine$maf,1-poag\_combine$maf)

poag\_combine$N<-c(rep(214634,nrow(poag\_combine)))



**# preparation of IOP data**

iop<-iop\[109:9007848,]

library(stringr)

library(tidyr)

colnames(iop)="NAME"

&nbsp;iop\_split<-str\_split\_fixed(iop$NAME, '\\t', 10)

iop\_split<-iop\_split\[,c(1:5,10)]

iop\_split<-data.frame(iop\_split)

iop\_split1<-iop\_split\[,6]

iop\_split1<-str\_split\_fixed(iop\_split$X6, ':', 5)

iop\_split1<-data.frame(iop\_split1)

&nbsp;iop\_combine<-cbind(iop\_split\[,1:5],iop\_split1)

iop\_combine<-iop\_combine\[,1:9]

colnames(iop\_combine)<-c('chr','pos','rsid','ref','alt',"beta","se","pval","maf")

iop\_combine<-iop\_combine\[-1,]

&nbsp;iop\_combine\[,1]<-as.numeric(iop\_combine\[,1])

iop\_combine\[,2]<-as.numeric(iop\_combine\[,2])

iop\_combine\[,6]<-as.numeric(iop\_combine\[,6])

iop\_combine\[,7]<-as.numeric(iop\_combine\[,7])

&nbsp;iop\_combine\[,8]<-as.numeric(iop\_combine\[,8])

&nbsp;iop\_combine\[,9]<-as.numeric(iop\_combine\[,9])

iop\_combine$pval<-10^(-iop\_combine$pval)

iop\_combine$maf<-ifelse(iop\_combine$maf>0.5,iop\_combine$maf,1-iop\_combine$maf)

iop\_combine$N<-c(rep(97653,nrow(iop\_combine)))



**# preparation of IS data**

is<-is\[112: 8279380,]

colnames(is)="NAME"

is\_split<-str\_split\_fixed(is$NAME, '\\t', 10)

is\_split<-is\_split\[,c(1:5,10)]

is\_split<-data.frame(is\_split)

is\_split1<-is\_split\[,6]

is\_split1<-str\_split\_fixed(is\_split$X6, ':', 5)

is\_split1<-data.frame(is\_split1)

is\_combine<-cbind(is\_split\[,1:5],is\_split1)

is\_combine<-is\_combine\[,1:9]

colnames(is\_combine)<-c('chr','pos','rsid','ref','alt',"beta","se","pval","maf")

is\_combine<-is\_combine\[-1,]

is\_combine\[,2]<-as.numeric(is\_combine\[,2])

is\_combine\[,6]<-as.numeric(is\_combine\[,6])

is\_combine\[,7]<-as.numeric(is\_combine\[,7])

is\_combine\[,8]<-as.numeric(is\_combine\[,8])

is\_combine\[,9]<-as.numeric(is\_combine\[,9])

is\_combine$pval<-10^(-is\_combine$pval)

is\_combine$maf<-ifelse(is\_combine$maf>0.5,is\_combine$maf,1-is\_combine$maf)

is\_combine$N<-c(rep(440328,nrow(is\_combine)))



**# MRlap study on ¡°IOP to IS¡±**

A = MRlap(exposure = iop\_combine,

&nbsp;                                           exposure\_name = "iop",

&nbsp;                                           outcome = is\_combine,

&nbsp;                                           outcome\_name = "is",

&nbsp;                                           ld = "./eur\_w\_ld\_chr",

&nbsp;                                           hm3 = "./w\_hm3.noMHC.snplist")

write.csv(unlist(A\[\["MRcorrection"]]),"iop\_is\_MRlap\_result1.csv")

write.csv(unlist(A\[\["LDSC"]]),"iop\_is\_MRlap\_result2.csv")



**# MRlap study on ¡°POAG to IS¡±**

B = MRlap(exposure = poag\_combine,

&nbsp;                                           exposure\_name = "poag",

&nbsp;                                           outcome = is\_combine,

&nbsp;                                           outcome\_name = "is",

&nbsp;                                           ld = "./eur\_w\_ld\_chr",

&nbsp;                                           hm3 = "./w\_hm3.noMHC.snplist")



write.csv(unlist(B\[\["MRcorrection"]]),"poag\_is\_MRlap\_result1.csv")

write.csv(unlist(B\[\["LDSC"]]),"poag\_is\_MRlap\_result2.csv")



**# 1.4 Multivariable mendelian randomization (MVMR)**

id\_exposure<-c('ukb-b-14146','finn-b-H7\_GLAUCPRIMOPEN')

exposure\_mv<-mv\_extract\_exposures(

&nbsp; id\_exposure,

&nbsp; clump\_r2 = 0.001,

&nbsp; clump\_kb = 10000,

&nbsp; harmonise\_strictness = 2,

&nbsp; find\_proxies = TRUE,

&nbsp; force\_server = FALSE,

&nbsp; pval\_threshold = 5e-08,

&nbsp; pop = "EUR"

)

mv\_outcome<-extract\_outcome\_data(exposure\_mv$SNP, 'ebi-a-GCST006908')

mv\_dat<-mv\_harmonise\_data(exposure\_mv, mv\_outcome, harmonise\_strictness = 2)

mv\_res<-mv\_multiple(

&nbsp;    mv\_dat,

&nbsp;    intercept = FALSE,

&nbsp;    instrument\_specific = FALSE,

&nbsp;    pval\_threshold = 5e-08,

&nbsp;    plots = FALSE

&nbsp;)

