**# 3. Performing Colocalization analysis**



**# 3.1 Analysis on gene CDKN2A**

library(TwoSampleMR)

library(MendelianRandomization)

library(ieugwasr)

library(stringr)

library(coloc)

chrpos <- paste0(9, ":",  21967751-(5E+5), "-", 21995323+(5E+5))

exposure\_1<-associations(variants=chrpos, id='finn-b-H7\_GLAUCPRIMOPEN', proxies=0)

outcome\_is<-associations(variants=chrpos, id=c("ebi-a-GCST006908"), proxies=0)

exposure\_1<-exposure\_1\[which(exposure\_1$rsid %in% unique(outcome\_is$rsid)),]

outcome\_is<-outcome\_is\[which(outcome\_is$rsid %in% unique(exposure\_1$rsid)),]

exposure\_1<-exposure\_1\[order(exposure\_1$p),] exposure\_1<-exposure\_1\[!duplicated(exposure\_1$rsid),]

outcome\_is<-outcome\_is\[order(outcome\_is$p),] outcome\_is<-outcome\_is\[!duplicated(outcome\_is$rsid),]

exposure\_1$n<-c(rep(214634,length(rownames(exposure\_1))))

res <- coloc.abf(dataset1 = list(beta=exposure\_1$beta,varbeta=exposure\_1$se^2,N=exposure\_1$n,type='quant',sdY=1,snp=exposure\_1$rsid),dataset2 = list(beta=outcome\_is$beta,varbeta=outcome\_is$se^2,N=outcome\_is$n,type='cc',snp=outcome\_is$rsid,MAF=outcome\_is$eaf))

res<-t(as.data.frame(res$summary))

write.csv(res,'poag\_CDKN2A\_IS\_coloc.csv')

write.table(exposure\_1,'exposure\_1.txt')

write.table(outcome\_is,'outcome\_is.txt')

locuscompare(in\_fn1='./exposure\_1.txt', in\_fn2='./outcome\_is.txt', title1="CDKN2A", title2="IS", marker\_col1="rsid", pval\_col1="p", marker\_col2="rsid", pval\_col2="p",population = "EUR",genome = c("hg19", "hg38"))



**# 3.2 Analysis on gene CDKN2A**

chrpos <- paste0(9, ":",  21994790-(5E+5), "-",22121096+(5E+5))

exposure\_1<-associations(variants=chrpos, id='finn-b-H7\_GLAUCPRIMOPEN', proxies=0)

outcome\_is<-associations(variants=chrpos, id=c("ebi-a-GCST006908"), proxies=0)

exposure\_1<-exposure\_1\[which(exposure\_1$rsid %in% unique(outcome\_is$rsid)),]

outcome\_is<-outcome\_is\[which(outcome\_is$rsid %in% unique(exposure\_1$rsid)),]

exposure\_1<-exposure\_1\[order(exposure\_1$p),] exposure\_1<-exposure\_1\[!duplicated(exposure\_1$rsid),]

outcome\_is<-outcome\_is\[order(outcome\_is$p),] outcome\_is<-outcome\_is\[!duplicated(outcome\_is$rsid),]

exposure\_1$n<-c(rep(214634,length(rownames(exposure\_1))))

res <- coloc.abf(dataset1 = list(beta=exposure\_1$beta,varbeta=exposure\_1$se^2,N=exposure\_1$n,type='quant',sdY=1,snp=exposure\_1$rsid),dataset2 = list(beta=outcome\_is$beta,varbeta=outcome\_is$se^2,N=outcome\_is$n,type='cc',snp=outcome\_is$rsid,MAF=outcome\_is$eaf))

res<-t(as.data.frame(res$summary))

write.csv(res,'poag\_CDKN2B-AS1\_IS\_coloc.csv')

write.table(exposure\_1,'exposure\_1\_cdkn2b-as1.txt')

write.table(outcome\_is,'outcome\_is\_cdkn2b-as1.txt')

locuscompare(in\_fn1='./exposure\_1\_cdkn2b-as1.txt', in\_fn2='./outcome\_is\_cdkn2b-as1.txt', title1="CDKN2B-AS1", title2="IS", marker\_col1="rsid", pval\_col1="p", marker\_col2="rsid", pval\_col2="p",population = "EUR",genome = c("hg19", "hg38"))



**# 3.3 Analysis on gene GAS7**

chrpos <- paste0(17, ":",  9813923-(5E+5), "-", 10101923+(5E+5))

exposure\_1<-associations(variants=chrpos, id='finn-b-H7\_GLAUCPRIMOPEN', proxies=0)

outcome\_is<-associations(variants=chrpos, id=c("ebi-a-GCST006908"), proxies=0)

exposure\_1<-exposure\_1\[which(exposure\_1$rsid %in% unique(outcome\_is$rsid)),]

outcome\_is<-outcome\_is\[which(outcome\_is$rsid %in% unique(exposure\_1$rsid)),]

exposure\_1<-exposure\_1\[order(exposure\_1$p),] exposure\_1<-exposure\_1\[!duplicated(exposure\_1$rsid),]

outcome\_is<-outcome\_is\[order(outcome\_is$p),] outcome\_is<-outcome\_is\[!duplicated(outcome\_is$rsid),]

&nbsp;exposure\_1$n<-c(rep(214634,length(rownames(exposure\_1))))

&nbsp;res <- coloc.abf(dataset1 = list(beta=exposure\_1$beta,varbeta=exposure\_1$se^2,N=exposure\_1$n,type='quant',sdY=1,snp=exposure\_1$rsid),dataset2 = list(beta=outcome\_is$beta,varbeta=outcome\_is$se^2,N=outcome\_is$n,type='cc',snp=outcome\_is$rsid,MAF=outcome\_is$eaf))

&nbsp;res<-t(as.data.frame(res$summary))

write.csv(res,'poag\_GAS7\_IS\_coloc.csv')

write.table(exposure\_1,'exposure\_1\_GAS7.txt')

write.table(outcome\_is,'outcome\_is\_GAS7.txt')

locuscompare(in\_fn1='./exposure\_1\_GAS7.txt', in\_fn2='./outcome\_is\_GAS7.txt', title1="GAS7", title2="IS", marker\_col1="rsid", pval\_col1="p", marker\_col2="rsid", pval\_col2="p",population = "EUR",genome = c("hg19", "hg38"))



