**# 5. Single-SNP mendelian randomization**

library(TwoSampleMR)

library(MendelianRandomization)

library(data.table)



**#5.1 download allsnps.txt.gz, cS2G\_UKBB.zip and cS2G\_1000GEUR.zip files from #¡°https://zenodo.org/record/7754032¡±**



**# import cS2G\_1000GEUR data** 

a=list.files('./cS2G\_1000GEUR')

dir = paste("./cS2G\_1000GEUR/",a,sep="")

n=length(dir)

merge.data =fread(file = dir\[1])

merge.data$name<-c(rep(a\[1],length(rownames(merge.data))))

for (i in 2:n){

&nbsp;         new.data = fread(file = dir\[i])

&nbsp;         new.data$name<-c(rep(a\[i],length(rownames(new.data))))

&nbsp;         merge.data = rbind(merge.data,new.data)

&nbsp;     }

snp<-fread('./allsnps.txt.gz',header = FALSE)

merge.data$chr<-snp\[match(merge.data$SNP,snp$V1),'V2']

merge.data$pos<-snp\[match(merge.data$SNP,snp$V1),'V3']



**# import cS2G\_UKBB data** 

a1=list.files('./cS2G\_UKBB')

dir1 = paste("./cS2G\_UKBB/",a,sep="")

n=length(dir1)

merge.data1 =fread(file = dir1\[1])

merge.data1$name<-c(rep(a1\[1],length(rownames(merge.data1))))

for (i in 2:n){

&nbsp;         new.data = fread(file = dir1\[i])

&nbsp;         new.data$name<-c(rep(a1\[i],length(rownames(new.data))))

&nbsp;         merge.data1 = rbind(merge.data1,new.data)

&nbsp;     }



snp<-fread('./allsnps\_ukbb.txt.gz',header = FALSE)



merge.data1$chr<-snp\[match(merge.data1$SNP,snp$V1),'V2']

merge.data1$pos<-snp\[match(merge.data1$SNP,snp$V1),'V3']



merge.data1$chr1<-substr(merge.data1$SNP,1,str\_locate(merge.data1$SNP,':')-1)

merge.data1$pos1<-substr(merge.data1$SNP,str\_locate(merge.data1$SNP,':')+1,str\_locate(merge.data1$SNP,'\_')-1)

merge.data1\[which(is.na(merge.data1$chr))]$chr<-merge.data1\[which(is.na(merge.data1$chr))]$'chr1'

merge.data1\[which(is.na(merge.data1$pos))]$pos<-merge.data1\[which(is.na(merge.data1$pos))]$'pos1'



**# 5.2 SSMR on IOP** 

**# download finn-b-H7\_GLAUCPRIMOPEN.vcf.gz and ukb-b-14146.vcf.gz files from the IEU GWAS #platform.** 



poag<-fread('./finn-b-H7\_GLAUCPRIMOPEN.vcf.gz', fill=TRUE)

iop<-fread('./ukb-b-14146.vcf.gz', fill=TRUE)



**# preparation of IOP data**

iop<-iop\[109:9007848,]

library(stringr)

library(tidyr)

colnames(iop)="NAME"

iop\_split<-str\_split\_fixed(iop$NAME, '\\t', 10)

iop\_split<-iop\_split\[,c(1:5,10)]

iop\_split<-data.frame(iop\_split)

iop\_split1<-iop\_split\[,6]

iop\_split1<-str\_split\_fixed(iop\_split$X6, ':', 5)

iop\_split1<-data.frame(iop\_split1)

iop\_combine<-cbind(iop\_split\[,1:5],iop\_split1)

iop\_combine<-iop\_combine\[,1:9]

colnames(iop\_combine)<-c('chr','pos','rsid','ref','alt',"beta","se","pval","maf")

iop\_combine<-iop\_combine\[-1,]

iop\_combine\[,1]<-as.numeric(iop\_combine\[,1])

iop\_combine\[,2]<-as.numeric(iop\_combine\[,2])

iop\_combine\[,6]<-as.numeric(iop\_combine\[,6])

iop\_combine\[,7]<-as.numeric(iop\_combine\[,7])

iop\_combine\[,8]<-as.numeric(iop\_combine\[,8])

iop\_combine\[,9]<-as.numeric(iop\_combine\[,9])

iop\_combine$pval<-10^(-iop\_combine$pval)

iop\_combine$maf<-ifelse(iop\_combine$maf>0.5,iop\_combine$maf,1-iop\_combine$maf)

iop\_combine$N<-c(rep(97653,nrow(iop\_combine)))



iop\_r2<-iop\_combine\[which(iop\_combine$pval<5E-8),]

iop\_r2$id.exposure<-iop\_r2$rsid

colnames(iop\_r2)<-c('chr.exposure','pos.exposure','SNP','effect\_allele.exposure','other\_allele.exposure','beta.exposure','se.exposure','pval.exposure','eaf.exposure','samplesize.exposure','id.exposure')

exposure\_dat4 <- format\_data(

&nbsp; iop\_r2,

&nbsp; type = "exposure",

&nbsp; snps = NULL,

&nbsp; header = TRUE,

&nbsp; snp\_col = "SNP",

&nbsp; beta\_col = "beta.exposure",

&nbsp; se\_col = "se.exposure",

&nbsp; eaf\_col = "eaf.exposure",

&nbsp; effect\_allele\_col = "effect\_allele.exposure",

&nbsp; other\_allele\_col = "other\_allele.exposure",

&nbsp; pval\_col = "pval.exposure",

&nbsp; samplesize\_col = "samplesize.exposure",

&nbsp; id\_col = "id.exposure",

&nbsp; min\_pval = 1e-310,

&nbsp; chr\_col = "chr.exposure",

&nbsp; pos\_col = "pos.exposure",

&nbsp; log\_pval = FALSE,phenotype\_col = 'id.exposure'

)



exposure\_dat6<-exposure\_dat4

&nbsp;chd\_out\_dat <- extract\_outcome\_data(

&nbsp;    snps = exposure\_dat6$SNP,

&nbsp;    outcomes = 'ebi-a-GCST006908',proxies = 1, rsq = 0.8, align\_alleles = 1, palindromes = 1, maf\_threshold = 0.3

&nbsp;)

dat <- harmonise\_data(

&nbsp;   exposure\_dat = exposure\_dat6, 

&nbsp;   outcome\_dat = chd\_out\_dat

)

res <- mr(dat)

res$start<-iop\_r2\[match(res$exposure,iop\_r2$SNP),]$'pos.exposure'

res$seqnames<-iop\_r2\[match(res$exposure,iop\_r2$SNP),]$'chr.exposure'

ssmr\_res\_005<-res\[res$pval<0.05,]

res1<-merge(ssmr\_res\_005,merge.data,by.x = c('seqnames','start'),by.y = c('chr','pos'))

res2<-merge(ssmr\_res\_005,merge.data1,by.x = c('seqnames','start'),by.y = c('chr','pos'))

res3<-merge(ssmr\_res\_005,merge.data,by.x = c('exposure'),by.y = c('SNP'))

res4<-merge(ssmr\_res\_005,merge.data1,by.x = c('exposure'),by.y = c('SNP'))

res3$SNP<-res3$exposure

res4$SNP<-res4$exposure

res2<-res2\[,colnames(res1)]

res3<-res3\[,colnames(res1)]

res4<-res4\[,colnames(res1)]

res\_comb<-rbind(res1,res2,res3,res4)



**# The similar process can be applied to POAG**



